lib/rucio/transfertool/fts3.py

```diff
@@ -60,17 +61,61 @@ def __init__(self, external_host):
 
         :param external_host:   The external host where the transfertool API is running
         """
+        usercert = config_get('conveyor', 'usercert', False, None)
 
+        self.deterministic_id = config_get_bool('conveyor', 'use_deterministic_id', False, False)
         super(FTS3Transfertool, self).__init__(external_host)
         if self.external_host.startswith('https://'):
-            self.cert = (USERCERT, USERCERT)
+            self.cert = (usercert, usercert)
             self.verify = False
         else:
             self.cert = None
             self.verify = True  # True is the default setting of a requests.* method
 
     # Public methods part of the common interface
 
+    def delegate_proxy(self, proxy, ca_path='/etc/grid-security/certificates/', duration_hours=48, timeleft_hours=12):
+        """Delegate user proxy to fts server if the lifetime is less than timeleft_hours
+
+        :param proxy: proxy to be delegated
+        :type proxy: str
+
+        :param ca_path: ca path for verification, defaults to '/etc/grid-security/certificates/'
+        :param ca_path: str, optional
+        :param duration_hours: delegation validity duration in hours, defaults to 48
+        :param duration_hours: int, optional
+        :param timeleft_hours: minimal delegation time left, defaults to 12
+        :param timeleft_hours: int, optional
+
+        :return: delegation ID
+        :rtype: str
+        """
+
+        logging.info("Delegating proxy %s to %s", proxy, self.external_host)
+
+        try:
+            context = Context(self.external_host,
```

Comment:
can you add a call to graphite with the core.monitor so we can watch when/what is being delegated

Reply:
need to take a look at core.monitor, but I see no problem in adding it